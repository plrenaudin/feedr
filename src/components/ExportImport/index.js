import db from "../../modules/db";
import { shortFormatDate } from "../../modules/formatter";
import Icon from "../Icon";

const exportData = () => {
  db
    .getAll()
    .then(data => {
      const dataStr = "data:text/json;charset=utf-8," + encodeURIComponent(JSON.stringify(data));
      const downloadAnchorNode = document.createElement("a");
      downloadAnchorNode.setAttribute("href", dataStr);
      downloadAnchorNode.setAttribute("download", `feedrExport-${shortFormatDate()}.json`);
      document.body.appendChild(downloadAnchorNode); // required for firefox
      downloadAnchorNode.click();
      downloadAnchorNode.remove();
    })
    .catch(e => console.error(e));
};

const onFileSelected = e => {
  const reader = new FileReader();
  reader.onload = () => importData(reader.result);
  reader.readAsText(e.target.files[0]);
};

const importData = data => {
  try {
    const parsed = JSON.parse(data);
    db.import(parsed).catch(e => console.error(e));
  } catch (e) {
    console.error(e);
  }
};

const ExportImport = () => (
  <div class="export-import">
    <div class="export" onClick={exportData}>
      <Icon name="download3" />
    </div>
    <div class="import">
      <label>
        <Icon name="upload3" />
        <input type="file" onChange={onFileSelected} />
      </label>
    </div>
  </div>
);

export default ExportImport;
