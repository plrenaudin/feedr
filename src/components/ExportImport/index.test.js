import { render } from "inferno";
import ExportImport from ".";

it("Matches the snapshot", () => {
  const div = document.createElement("div");
  expect(render(<ExportImport />, div).dom).toMatchSnapshot();
});
