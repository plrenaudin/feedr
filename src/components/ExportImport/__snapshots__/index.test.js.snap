// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1`] = `
<div
  class="export-import"
>
  <div
    class="export"
  >
    <span
      class="icon-wrapper"
    >
      <svg
        class="icon"
      >
        <use
          xlink:href="#icon-download3"
        />
      </svg>
    </span>
  </div>
  <div
    class="import"
  >
    <label>
      <span
        class="icon-wrapper"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="#icon-upload3"
          />
        </svg>
      </span>
      <input
        type="file"
      />
    </label>
  </div>
</div>
`;
